data/global.lua

```diff
@@ -231,3 +231,24 @@ function getPlayerDatabaseInfo(name_or_guid)
 	result.free(query)
 	return info
 end
+
+-- Function to check for duplicate keys in a table
+-- Receives the name of the table to be checked as argument
+function checkDuplicatesValues(varName)
+	-- Retrieve the table to be checked
+	local keys = _G[varName]
+	-- Create a table to keep track of the keys already seen
+	local seen = {}
+	-- Iterate over the keys in the table
+	for k, v in pairs(keys) do
+		-- Check if a key has already been seen
+		if seen[v] then
+			-- If it has, return true and the duplicate key
+			return true, "Duplicate key found: " .. v
```

Comment:
Don't return it, and let it find all duplicates before returning. That way, if the key is duplicated more than once, it will be caught at the same time instead.

Given that you are printing the message, maybe return a list of duplicate values and build the message at call point.

Reply:
done!